File: Day 73/203. Remove Linked List Elements.py
# Definition for singly-linked list.
class ListNode:
    def __init__(self, val=0, next=None):
        self.val = val
        self.next = next

class Solution:
    def removeElements(self, head: Optional[ListNode], val: int) -> Optional[ListNode]:

        temp = head

        while(temp is not None and temp.next is not None):
            curr = temp.next
            
            if(curr.val == val):
                temp.next = curr.next
                curr = None

            else:
                temp = temp.next

        temp = head
        
        if(temp):
            if(temp.val == val):
                return temp.next

        return head

File: Day 73/203. Remove Linked List Elements.cpp

  struct ListNode {
      int val;
      ListNode *next;
      ListNode() : val(0), next(nullptr) {}
      ListNode(int x) : val(x), next(nullptr) {}
     ListNode(int x, ListNode *next) : val(x), next(next) {}
  };

class Solution {
public:
    ListNode* removeElements(ListNode* head, int val) {
        
        ListNode* temp = head;

        while(temp != nullptr && temp -> next != nullptr){
            ListNode* curr = temp -> next;
            
            if(curr -> val == val){
                temp -> next = curr -> next;
                delete (curr);
            }
            

            else temp = temp -> next;
        }

        temp = head;
        
        if(temp){
            if(temp -> val == val) return temp -> next;
        }

        return head;
    }
};